src/ResourceManager/IotHub/Commands.IotHub.Test/ScenarioTests/IotHubTests.ps1

```diff
@@ -226,6 +226,43 @@ function Test-AzureRmIotHubLifecycle
 	$iotHubUpdated = Set-AzureRmIotHub -ResourceGroupName $ResourceGroupName -Name $IotHubName -FallbackRoute $iothub.Properties.Routing.FallbackRoute	
     Assert-True { $iotHubUpdated.Properties.Routing.FallbackRoute.IsEnabled -eq 1}
 
+	# Add Certificate
+	$certificatePath = "ScenarioTestFile\PS-Test-Root-CA.cer"
+	$certificateSubject = "CN=PS Test Root Certificate Authority"
+	$certificateThumbprint = "38303FC7371EC78DDE3E18D732C8414EE50969C7"
+	$certificateType = "Microsoft.Devices/IotHubs/Certificates"
+	$certificateName = "TestCertificate"
+	$newCertificate = Add-AzureRmIotHubCertificate -ResourceGroupName $ResourceGroupName -Name $IotHubName -CertificateName $certificateName -Path $certificatePath
+	Assert-True { $newCertificate.Properties.Subject -eq $certificateSubject }
+	Assert-True { $newCertificate.Properties.Thumbprint -eq $certificateThumbprint }
+	Assert-False { $newCertificate.Properties.IsVerified }
+	Assert-True { $newCertificate.Type -eq $certificateType }
+	Assert-True { $newCertificate.CertificateName -eq $certificateName }
+
+	# List All Certificate
+	$certificates = Get-AzureRmIotHubCertificate -ResourceGroupName $ResourceGroupName -Name $IotHubName
+	Assert-True { $certificates.Count -gt 0}
+
+	# Get Certificate
+	$certificate = Get-AzureRmIotHubCertificate -ResourceGroupName $ResourceGroupName -Name $IotHubName -CertificateName $certificateName
+	Assert-True { $certificate.Properties.Subject -eq $certificateSubject }
+	Assert-True { $certificate.Properties.Thumbprint -eq $certificateThumbprint }
+	Assert-False { $certificate.Properties.IsVerified }
+	Assert-True { $certificate.Type -eq $certificateType }
+	Assert-True { $certificate.CertificateName -eq $certificateName }
+
+	# Get Verification Code
+	$certificateWithNonce = Get-AzureRmIotHubCertificateVerificationCode -ResourceGroupName $ResourceGroupName -Name $IotHubName -CertificateName $certificateName -Etag $certificate.Etag
+	Assert-True { $certificateWithNonce.Properties.Subject -eq $certificateSubject }
+	Assert-NotNull { $certificateWithNonce.Properties.VerificationCode }
```

Comment:
Where is the test for Set-AzureRmIotHubVerifiedCertificate?

Reply:
Can't make unit test for this cmdlet because it requires the proof-of-possession flow

Reply:
Why can't you run the proof of possession script in the test?

Reply:
You should be able to do the same steps as in [this script](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-security-x509-create-certificates#proof-of-possession-of-your-x509-ca-certificate) with powershell cmdlets built into PowerShell 4+.

Reply:
added requested test but the test will execute in admin mode because of using New-SelfSignedCertificate to create certificate at runtime and it executes in admin mode only